neqo-bin/src/client/mod.rs

```diff
@@ -528,19 +532,29 @@ impl<'a, H: Handler> Runner<'a, H> {
     }
 }
 
+#[cfg_attr(
+    not(feature = "qlog"),
+    expect(unused_variables, reason = "Only used with qlog.")
+)]
 fn qlog_new(args: &Args, hostname: &str, cid: &ConnectionId) -> Res<Qlog> {
```

Comment:
You could use

```rust
if cfg!(not(feature = "qlog")) {
  return Err(Error::Argument("qlog...");
}
```

I think that avoids all the unused variable warnings and keeps the code simple.

This crate, like most of the binary crates (and test-fixture), isn't directly exposed to the qlog crate, so it should be possible to go through the motions of setting up logging, which would do nothing if the neqo-common crate had the feature disabled.

The -transport and -http crates need to be exposed to qlog types, which means that it makes sense for disabling there to be more invasive.  However, there might be a different approach.

Reply:
That gets us an `unreachable_code` warning...